core-ui/src/shared/components/MonacoEditorESM/Editor.js

```diff
@@ -50,6 +55,37 @@ export function Editor({
     memoizedOptions.current = options;
   }
 
+  useEffect(() => {
+    if (editorRef.current) {
+      focusRef.current?.dispose();
+      focusRef.current = editorRef.current.onDidBlurEditorText(() => {
+        setHasFocus(false);
+        if (typeof onBlur === 'function') {
+          onBlur();
+        }
+      });
+      blurRef.current?.dispose();
+      blurRef.current = editorRef.current.onDidFocusEditorText(() => {
+        setHasFocus(true);
+        if (typeof onFocus === 'function') {
+          onFocus();
+        }
+      });
+      return () => {
+        focusRef.current.dispose();
+        blurRef.current.dispose();
+      };
+    }
+  }, [onBlur, onFocus, editorRef]);
```

Comment:
onBlur is recreated with new identity on every render, hence this code is executed on every render. If it's supposed to work that way, it shouldn't be placed in useEffect.

Reply:
That's true, but it wouldn't be a case if someone wraps `onBlur` in `useCallback`, right?